### PR TITLE
Add search functionality.

### PR DESCRIPTION
**Pull request by Acidity**
Part of #192.

### CODE REVIEW

**Comment by jboning** [comment]
I'd suggest breaking up the search input UI and endpoint to be separate forms and endpoints. Clearer for the user that you can only search one kind of thing at a time, and I think clearer reading the code, too.

File: brave/core/admin/controller.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+class AdminController(Controller):
+    """Entry point for the Search RESTful interface."""
+
+    #index = AdminInterface()
+
+    def index(self):
+        return 'brave.core.admin.template.search', dict(area='admin')
+    
+    def search(self, Character=None, charMethod=None, Alliance=None, Corporation=None, group=None, KeyID=None,
```
Comment (jboning) [inline brave/core/admin/controller.py]: curious why the uppercased characters (usually uppercase is used to indicate class names in python)

also, "group" here does not match "Group" in the template

**Comment by jboning** [comment]
oh, also, the select inputs for how to match character/username are kind of confusing

**Comment by jboning** [comment]
Did you mean to add new template files?

I did not mean to suggest breaking up the search forms onto separate pages, sorry :( just making the difference search types visibly different forms. Don't feel obligated to change it though, I'm happy to try some different designs myself after landing.

**Comment by Acidity** [comment]
errr, yeah I forgot the templates...

Also, I think the separate pages works better from a design standpoint now that I see it when it's actually done, but that may just be me.

**Pull request merged**